### ISSUE TITLE
[ts] 'privateFunction' is declared but its value is never read.

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#25232]
_From @francofslab15 on June 26, 2018 15:33_

Got here and found an issue similar to: https://github.com/Microsoft/vscode/issues/51814 which was marked as not reproducible.

Verified that the insiders build has the same issue as the current release. So I used it to test it without any external extensions.

- VSCode Version: 1.25.0-insider
- OS Version: Windows 10 Pro - Version 1803 - Build 17134.112

Steps to Reproduce:
Using latest TypeScript npm package, create the typescript class below:
```ts
export class Test {
    private _someVar: boolean = false; //[ts] '_someVar' is declared but its value is never read.

    public somePublicFunction = function() {
        var lib = new ExternalLibrary();
        this._somePrivateFunction();
    }

    private _somePrivateFunction = function() { //[ts] '_somePrivateFunction' is declared but its value is never read.
        if (this._someVar) { //_someVar is read here.
            console.log(':/');
        }
    }
}
```

Does this issue occur when all extensions are disabled?: Yes

This happens only when we declare class functions as anonymous. Although not a best coding practice, VSCode is still flagging those variables or functions incorrectly.

_Copied from original issue: Microsoft/vscode#52997_

**Comment by RyanCavanaugh** [comment]
`this` in these functions is `any` because those aren't arrow functions, so we don't have any way to realize that `_somePrivateFunction` is being used. You can add an explicit `this` type if needed:

        public somePublicFunction = function(this: Test) {

**Comment by francofslab15** [comment]
The funny thing is that TSLint does not catch it with the "no-unused-variable" rule. So there must be a way to determine that. Anyways, I already fixed on my side to disable the warning, however, wanted to make sure TypeScript/VSCode was working as intended.

Anyways, thanks for the suggestion.

Cheers.
